tests/rptest/scale_tests/high_throughput_test.py

```diff
@@ -0,0 +1,928 @@
+
+        # create a new topic with low local retention.
+        config = {
+            'segment.bytes': segment_size,
+            'retention.bytes': -1,
+            'retention.local.target.bytes': 2 * segment_size,
+            'cleanup.policy': 'delete',
+            'partition_autobalancing_node_availability_timeout_sec':
+            self.unavailable_timeout,
+            'partition_autobalancing_mode': 'continuous',
+            'raft_learner_recovery_rate': 10 * 1024 * 1024 * 1024,
+        }
+        self.rpk.create_topic(self.topic_name,
+                              partitions=self.scaled_num_partitions,
+                              replicas=3,
+                              config=config)
+
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       self.topic_name,
+                                       msg_size=self.msg_size,
+                                       msg_count=5 * 1024 * 1024 * 1024 * 1024,
+                                       rate_limit_bps=self.scaled_data_bps)
+        producer.start()
+
+        # produce 10 mins worth of consume data onto S3.
+        produce_time_s = 4 * 60
+        messages_to_produce = (produce_time_s * consume_rate) / self.msg_size
+        time_to_wait = (messages_to_produce *
+                        self.msg_size) / self.scaled_data_bps
+
+        wait_until(
+            lambda: producer.produce_status.acked >= messages_to_produce,
+            timeout_sec=1.5 * time_to_wait,
+            backoff_sec=5,
+            err_msg=
+            f"Could not ack production of {messages_to_produce} messages in {1.5 * time_to_wait} s"
+        )
+        # continue to produce the rest of the test
+
+        validator_overrides = {
+            OMBSampleConfigurations.E2E_LATENCY_50PCT:
+            [OMBSampleConfigurations.lte(51)],
+            OMBSampleConfigurations.E2E_LATENCY_AVG:
+            [OMBSampleConfigurations.lte(145)],
+        }
+
+        # Run a usual producer + consumer workload and a S3 producer + consumer workload concurrently
+        # Ensure that the S3 workload doesn't effect the usual workload majorly.
+        benchmark = self._run_omb(self.scaled_data_bps / 2,
+                                  validator_overrides)
+
+        # This consumer should largely be reading from S3
+        consumer = RpkConsumer(self._ctx,
+                               self.redpanda,
+                               self.topic_name,
+                               offset="oldest",
+                               num_msgs=messages_to_produce)
+        consumer.start()
+        wait_until(
+            lambda: consumer.message_count >= messages_to_produce,
+            timeout_sec=5 * produce_time_s,
+            backoff_sec=5,
+            err_msg=
+            f"Could not consume {messages_to_produce} msgs in {5 * produce_time_s} s"
+        )
```

Comment:
Just curious if you noticed how long this takes. Is it pretty quick? I wonder if we can do this repeatedly, given the segment size is pretty large and downloads may not be too frequent. Fine as is tho

Reply:
Good question. This one is Brandon's, I never ran it in a cluster of x4gen.4xlarges that it requires, and CDT can't stand it. Maybe @ballard26 knows?